api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/freemarker/DeleteIndividualController.java

```diff
@@ -0,0 +1,200 @@
+import edu.cornell.mannlib.vitro.webapp.controller.VitroRequest;
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.responsevalues.RedirectResponseValues;
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.responsevalues.ResponseValues;
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.responsevalues.TemplateResponseValues;
+import edu.cornell.mannlib.vitro.webapp.dao.DisplayVocabulary;
+import edu.cornell.mannlib.vitro.webapp.dao.jena.QueryUtils;
+import edu.cornell.mannlib.vitro.webapp.dao.jena.event.BulkUpdateEvent;
+import edu.cornell.mannlib.vitro.webapp.modelaccess.ModelNames;
+import edu.cornell.mannlib.vitro.webapp.rdfservice.ChangeSet;
+import edu.cornell.mannlib.vitro.webapp.rdfservice.RDFService;
+import edu.cornell.mannlib.vitro.webapp.rdfservice.RDFServiceException;
+import edu.cornell.mannlib.vitro.webapp.rdfservice.impl.RDFServiceUtils;
+
+@WebServlet(name="DeleteIndividualController",urlPatterns="/deleteIndividualController")
+public class DeleteIndividualController extends FreemarkerHttpServlet{
+	
+  private static final Log log = LogFactory.getLog(DeleteIndividualController.class);
+  private static final boolean BEGIN = true;
+  private static final boolean END = !BEGIN;
+	
+  private static String TYPE_QUERY_START = ""
+      + "PREFIX vitro:    <http://vitro.mannlib.cornell.edu/ns/vitro/0.7#>"
+      + "SELECT ?type "
+      + "WHERE"
+      + "{ <";
+  private static String TYPE_QUERY_END = "> vitro:mostSpecificType ?type ." 
+      +	"}";
+  private static String queryForDeleteQuery = 
+ 		    "PREFIX display: <" + DisplayVocabulary.DISPLAY_NS +"> \n" +
+        "SELECT ?deleteQueryText WHERE { ?associatedURI display:hasDeleteQuery ?deleteQueryText }";
+
+  private static final String DEFAULT_DELETE_QUERY_TEXT = "DESCRIBE ?individualURI";
+
+  @Override
+  protected AuthorizationRequest requiredActions(VitroRequest vreq) {
+    return SimplePermission.DO_FRONT_END_EDITING.ACTION;
+  }
+	
+  protected ResponseValues processRequest(VitroRequest vreq) {
+    String errorMessage = handleErrors(vreq);
+    if (!errorMessage.isEmpty()) {
+      return prepareErrorMessage(errorMessage);
+    }
+    String individualUri = vreq.getParameter("individualUri");
+    String type = getObjectMostSpecificType(individualUri, vreq);
+    Model displayModel = vreq.getDisplayModel();
+
+    String delteQueryText = getDeleteQueryForType(type, displayModel);
+    byte[] toRemove = getIndividualsToDelete(individualUri, delteQueryText, vreq);
+    if (toRemove.length > 0) {
+      deleteIndividuals(toRemove, vreq);
+    }
+    String redirectUrl = getRedirectUrl(vreq);
+
+    return new RedirectResponseValues(redirectUrl, HttpServletResponse.SC_SEE_OTHER);
+  }
+	
+  private String getRedirectUrl(VitroRequest vreq) {
+    String redirectUrl = vreq.getParameter("redirectUrl");
+    if (redirectUrl != null) {
+      return redirectUrl;
+    }
+    return "/";
```

Comment:
Can confirmDeleteIndividualForm.ftl (lines 5-9) be simplified? Do we need if clause there, or 
<#assign redirectUrl = editConfiguration.pageData.redirectUrl />
 is enough? Or we can simplify the code here in the Java file?

Reply:
I think validation is needed as it get values from the browser in both cases. Assigning limits configuration of redirect url. I would leave it that way.